lib/appmap/hook/method.rb

```diff
@@ -100,11 +102,34 @@ def activate
         hook_method_def = hook_method_def.ruby2_keywords if hook_method_def.respond_to?(:ruby2_keywords)
 
         hook_method_parameters = hook_method.parameters.dup.freeze
+        SIGNATURES[[ hook_class, hook_method.name ]] = hook_method_parameters
+
         hook_class.ancestors.first.tap do |cls|
           cls.define_method_with_arity(hook_method.name, hook_method.arity, hook_method_def)
-          redefined_method = cls.instance_method(hook_method.name)
-          redefined_method.singleton_class.module_eval do
-            define_method(:parameters) { hook_method_parameters }
+
+          lookup_method = lambda do |id|
+            method = super(id)
+
+            signature = SIGNATURES[[ method.owner, method.name ]]
+            if signature
+              method.singleton_class.module_eval do
+                define_method(:parameters) do
+                  signature
+                end
+              end
+            end
+
+            method
+          end
+
+          cls.module_eval do
```

Comment:
I don't think I need to do this over and over; just once on the base type. Right?